Sources/Workspace/Workspace.swift

```diff
@@ -1144,6 +1144,12 @@ extension Workspace {
         let contents = try fileSystem.readFileContents(path)
         return self.checksumAlgorithm.hash(contents).hexadecimalRepresentation
     }
+
+    public func fileAffectsSwiftOrClangBuildSettings(filePath: AbsolutePath, packageGraph: PackageGraph) -> Bool {
```

Comment:
packageGraph used?

Reply:
I should have mentioned this comment from @neonichu: https://github.com/apple/sourcekit-lsp/pull/443#discussion_r782419222. He suggested that we also pass the package graph so SwiftPM can provide a more sophisticated implementation in the future (e.g. based on the sources directories).

Reply:
ah, worth adding this code comment here for future maintainers?

Reply:
Good idea. Done.